### DISCUSSION TITLE
How to choose the matching icon color for menu background

### DISCUSSION THREAD

**Discussion by manuel-koch** [fyne-io/systray#70] [Q&A] [answered]
On MacOS the color of the main menu depends sometimes on the color of the background image of the display.
Thus if you choose e.g. a solid light gray color, then standard menu entries of MacOS appear in black.
But if you choose e.g. a solid dark gray color, then the standard menu entries appear in white.

Any idea how I can adjust my systray app to achieve the same for the main icon selected via call to `systray.SetIcon(...)` ?
I.e. I already have one icon in black and one in white - but I don't have a trigger when to choose which ?
Is there already something similar built into systray library ?

**Comment by andydotxyz** [comment]
Does this correlate with the users chosen colour scheme (light vs dark)? In which case you can get the "variant" from App.Settings().

**Reply by manuel-koch** [reply]
I'm just using `fyne.systray` and AFAIK there are no app settings available there ( like in the full fyne lib ).

Additionally reading Apple Resources, they are talking about some "themed icon" that uses special black/white color with transparency that will result in automatically switching between white and black appearance in the menu, because of automatic tinting. E.g. this [apple forum](https://developer.apple.com/forums/thread/662322) relates to the topic.

The issue is not just letting user select light/dark theme, the menu appears different when e.g. nightshift is enabled or the display background uses dark/light image and this can happen during runtime of the systray app.

I just found this method, which sounds like what I was looking for: `systray.SetTemplateIcon()`.
I will give it a try.

**Reply by manuel-koch** [reply, marked as answer]
Ok `systray.SetTemplateIcon()` is the method I was searching for. If I use a black/transparent PNG, the menu icon will automatically adjust to black/white appearance in the menu bar like other standard MacOS menu items.
